docs/data-structures.md

```diff
@@ -0,0 +1,249 @@
+            "index": 4294967295
+        },
+        "unlock": {
+            "args": [],
+            "binary": "0x0b00000000000000",
+            "reference": null,
+            "signed_args": [],
+            "version": 0
+        }
+        }
+    ],
+    "outputs": [
+        {
+        "capacity": 5000000,
+        "data": "0x",
+        "lock": "0xa58a960b28d6e283546e38740e80142da94f88e88d5114d8dc91312b8da4765a",
+        "type": null
+        }
+    ],
+    "version": 0
+    }
+],
+"header": {
+    "cellbase_id": "0xabeb06aea75b59ec316db9d21243ee3f0b0ad0723e50f57761cef7e07974b9b5",
+    "difficulty": "0x100",
+    "hash": "0xcddd882eff5edd2f7db25074cbbdc1d21cd698f60d6fb39412ef91d19eb900e8",
+    "number": 11,
+    "parent_hash": "0x255f65bf9dc00bcd9f9b8be8624be222cba16b51366208a8267f1925eb40e7e4",
+    "seal": {
+        "nonce": 503529102265201399,
+        "proof": "0x"
+    },
+    "timestamp": 1551155125985,
+    "txs_commit": "0xabeb06aea75b59ec316db9d21243ee3f0b0ad0723e50f57761cef7e07974b9b5",
+    "txs_proposal": "0x0000000000000000000000000000000000000000000000000000000000000000",
+    "uncles_count": 1,
+    "uncles_hash": "0x99cf8710e59303bfac236b57256fcea2c58192f2c9c39d1ea4c19cbcf88b4952",
+    "version": 0
+},
+"proposal_transactions": [],
+"uncles": [
+    {
+    "cellbase": {
+        ...
+    },
+    "header": {
+        ...
+    },
+    "proposal_transactions": []
+    }
+]
+}
+```
+
+### Description
+
+#### Block
+
+| Name                    | Type            | Description                                                  |
+| ----------------------- | --------------- | ------------------------------------------------------------ |
+| `header`                | `Header`        | **The block header of the block.** This part contains some metadata of the block. |
+| `commit_trasactions`    | [`Transaction`] | **An array of transactions contained in the block.** This is where the miner put the received transactions. |
+| `proposal_transactions` | [string]        | **An array of hex-encoded short transaction ID.**            |
+| `uncles`                | [`UncleBlock`]  | **An array of uncle blocks of the block.**                   |
+
+#### Header
+
+| Name           | Type                | Description                                                  |
+| -------------- | ------------------- | ------------------------------------------------------------ |
+| `cellbase_id`  | H256(hash)          | **The hash of the Cellbase transaction.** Cellbase transaction is just like the coinable transaction in Bitcoin. It's the transaction added by the miner who mined this block, by which the miner receives block reward for successfully mined the block. |
+| `difficulty`   | Bytes               | **The difficulty of the PoW puzzle.**                        |
+| `hash`         | H256(hash)          | **The block hash.**                                          |
```

Comment:
RPC returns this, but indeed it is not a field of the header, it is computed from the header.